diff --git a/meson.build b/meson.build
index 2d9200529ca..d738ed797e3 100644
--- a/meson.build
+++ b/meson.build
@@ -1454,11 +1454,11 @@ endif
 
 with_c11_threads = false
 if cc.has_function('thrd_create', prefix: '#include <threads.h>')
-  if with_platform_android
+  #if with_platform_android   # Enable MUSL
     # Current only Android's c11 <threads.h> are verified
     pre_args += '-DHAVE_THRD_CREATE'
     with_c11_threads = true
-  endif
+  #endif
 endif
 
 if cc.has_header_symbol('errno.h', 'program_invocation_name',
diff --git a/src/compiler/glsl/meson.build b/src/compiler/glsl/meson.build
index d2fa8b1810f..cc823055c9c 100644
--- a/src/compiler/glsl/meson.build
+++ b/src/compiler/glsl/meson.build
@@ -7,7 +7,7 @@ libglsl_util = static_library(
   include_directories : [inc_include, inc_src, inc_mesa, inc_gallium, inc_gallium_aux],
   dependencies : [dep_valgrind, idep_mesautil],
   gnu_symbol_visibility : 'hidden',
-  build_by_default : false,
+  build_by_default : true,
 )
 
 subdir('glcpp')
@@ -215,7 +215,7 @@ libglsl = static_library(
   link_with : libglcpp,
   include_directories : [inc_include, inc_src, inc_mesa, inc_gallium, inc_gallium_aux],
   dependencies : [idep_nir, idep_mesautil],
-  build_by_default : false,
+  build_by_default : true,
 )
 
 idep_libglsl = declare_dependency(
diff --git a/src/compiler/meson.build b/src/compiler/meson.build
index 92e72a44f9a..3dba7b489b2 100644
--- a/src/compiler/meson.build
+++ b/src/compiler/meson.build
@@ -51,7 +51,7 @@ libcompiler = static_library(
   cpp_args : [cpp_msvc_compat_args],
   gnu_symbol_visibility : 'hidden',
   dependencies : [dep_valgrind],
-  build_by_default : false,
+  build_by_default : true,
 )
 
 idep_compiler = declare_dependency(
diff --git a/src/compiler/nir/meson.build b/src/compiler/nir/meson.build
index 9320ce2d277..97a74707e0f 100644
--- a/src/compiler/nir/meson.build
+++ b/src/compiler/nir/meson.build
@@ -369,7 +369,7 @@ _libnir = static_library(
   c_args : [c_msvc_compat_args, no_override_init_args, no_misleading_indentation],
   gnu_symbol_visibility : 'hidden',
   dependencies : [idep_compiler, dep_valgrind, idep_mesautil],
-  build_by_default : false,
+  build_by_default : true,
 )
 
 inc_nir = include_directories('.')
diff --git a/src/egl/drivers/dri2/egl_dri2.c b/src/egl/drivers/dri2/egl_dri2.c
index 322dc3f790b..70261f2f2d7 100644
--- a/src/egl/drivers/dri2/egl_dri2.c
+++ b/src/egl/drivers/dri2/egl_dri2.c
@@ -2849,7 +2849,7 @@ dri2_create_sync(_EGLDisplay *disp, EGLenum type, const EGLAttrib *attrib_list)
       }
 #endif
 
-      ret = pthread_cond_init(&dri2_sync->cond, &attr);
+      ret = pthread_cond_init((pthread_cond_t*)&dri2_sync->cond, &attr);
 
       if (ret) {
          _eglError(EGL_BAD_ACCESS, "eglCreateSyncKHR");
diff --git a/src/egl/meson.build b/src/egl/meson.build
index b9baa09122c..687c82826c5 100644
--- a/src/egl/meson.build
+++ b/src/egl/meson.build
@@ -200,7 +200,7 @@ egl_def = custom_target(
   command : gen_vs_module_defs_normal_command,
 )
 
-libegl = shared_library(
+libegl = static_library(
   egl_lib_name,
   files_egl,
   c_args : [
@@ -215,10 +215,10 @@ libegl = shared_library(
   link_depends : [link_deps_for_egl],
   dependencies : [deps_for_egl, dep_dl, dep_libdrm, dep_clock, dep_thread, idep_mesautil],
   install : true,
-  version : egl_lib_version,
-  soversion : egl_lib_soversion,
+#  version : egl_lib_version,
+#  soversion : egl_lib_soversion,
   name_prefix : host_machine.system() == 'windows' ? 'lib' : [],  # always use lib, but avoid warnings on !windows
-  vs_module_defs : egl_def
+#  vs_module_defs : egl_def
 )
 
 if with_glvnd
diff --git a/src/gallium/auxiliary/meson.build b/src/gallium/auxiliary/meson.build
index b1dad4fce88..c523f6fc4e3 100644
--- a/src/gallium/auxiliary/meson.build
+++ b/src/gallium/auxiliary/meson.build
@@ -505,7 +505,7 @@ libgallium = static_library(
     dep_libdrm, dep_llvm, dep_dl, dep_m, dep_thread, dep_lmsensors, dep_ws2_32,
     idep_nir, idep_nir_headers, idep_mesautil,
   ],
-  build_by_default : false
+  build_by_default : true
 )
 
 if with_tests
diff --git a/src/gallium/auxiliary/pipe-loader/meson.build b/src/gallium/auxiliary/pipe-loader/meson.build
index 2bda1d28872..c76e50ee990 100644
--- a/src/gallium/auxiliary/pipe-loader/meson.build
+++ b/src/gallium/auxiliary/pipe-loader/meson.build
@@ -57,5 +57,5 @@ libpipe_loader_static = static_library(
   gnu_symbol_visibility : 'hidden',
   link_with : [libpipe_loader_links],
   dependencies : [dep_libdrm, idep_xmlconfig, idep_mesautil],
-  build_by_default : false,
+  build_by_default : true,
 )
diff --git a/src/gallium/targets/dri/meson.build b/src/gallium/targets/dri/meson.build
index 134d7e4adf8..6760c6932d6 100644
--- a/src/gallium/targets/dri/meson.build
+++ b/src/gallium/targets/dri/meson.build
@@ -37,7 +37,7 @@ else
   libgallium_name = 'gallium-@0@'.format(meson.project_version())
 endif
 
-libgallium_dri = shared_library(
+libgallium_dri = static_library(
   libgallium_name,
   files('dri_target.c'),
   include_directories : [
diff --git a/src/gbm/backends/dri/meson.build b/src/gbm/backends/dri/meson.build
index 5c33a3a907e..71a089aca62 100644
--- a/src/gbm/backends/dri/meson.build
+++ b/src/gbm/backends/dri/meson.build
@@ -12,14 +12,14 @@ if with_platform_wayland
   endif
 endif
 
-shared_library(
+static_library(
   'dri_gbm',
   files('gbm_dri.c', 'gbm_driint.h'),
   include_directories : [inc_gallium, incs_gbm_dri, inc_loader, inc_st_dri, inc_gallium_aux],
   c_args : [flags_gbm_dri],
   link_args : [ld_args_gc_sections],
   link_with : [libloader, libgallium_dri],
-  dependencies : [deps_gbm_dri, dep_dl, dep_gbm, dep_libdrm, idep_mesautil, idep_xmlconfig],
+  dependencies : [deps_gbm_dri, dep_dl, dep_gbm, dep_libdrm, idep_mesautil, idep_xmlconfig, dep_spirv_tools],
   gnu_symbol_visibility : 'hidden',
   install : true,
   install_dir: join_paths(get_option('libdir'), 'gbm'),
diff --git a/src/gbm/meson.build b/src/gbm/meson.build
index 97e8d5fa044..e1ae56c343b 100644
--- a/src/gbm/meson.build
+++ b/src/gbm/meson.build
@@ -21,7 +21,7 @@ if with_platform_android and get_option('platform-sdk-version') >= 30
   libgbm_name = 'gbm_mesa'
 endif
 
-libgbm = shared_library(
+libgbm = static_library(
   libgbm_name,
   files_gbm,
   include_directories : incs_gbm,
@@ -30,7 +30,7 @@ libgbm = shared_library(
   link_with : [libloader],
   dependencies : [dep_libdrm, idep_xmlconfig],
   gnu_symbol_visibility : 'hidden',
-  version : '1.0.0',
+#  version : '1.0.0',
   install : true,
 )
 
diff --git a/src/loader/loader.c b/src/loader/loader.c
index 15bdcbfff07..56b6d2b80e6 100644
--- a/src/loader/loader.c
+++ b/src/loader/loader.c
@@ -37,6 +37,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <limits.h>
+#include <libgen.h>
 #include <sys/param.h>
 #ifdef MAJOR_IN_MKDEV
 #include <sys/mkdev.h>
@@ -872,9 +873,35 @@ loader_open_driver_lib(const char *driver_name,
          next = end;
 
       len = next - p;
-      snprintf(path, sizeof(path), "%.*s/%s%s.so", len,
-               p, driver_name, lib_suffix);
-      driver = dlopen(path, RTLD_NOW | RTLD_LOCAL);
+
+
+      // if (strcmp(driver_name, "dri") == 0) {
+      if (strncmp(p, "<dirname>", len) == 0) {
+
+         char exe_path[PATH_MAX];
+         ssize_t count = readlink("/proc/self/exe", exe_path, PATH_MAX - 1);
+
+         if (count != -1) {
+            // Successfully read the link; null-terminate the buffer
+            exe_path[count] = 0;
+            printf("exe_path: %s\n", exe_path);
+            p = dirname(exe_path);
+            printf("dirname: %s\n", p);
+
+            snprintf(path, sizeof(path), "%s/%s%s.so",
+                     p, driver_name, lib_suffix);
+         } else {
+            // if /proc/self/exe is not available)
+            continue;
+         }
+      } else {
+         snprintf(path, sizeof(path), "%.*s/%s%s.so", len,
+                  p, driver_name, lib_suffix);
+      }
+
+//      driver = dlopen(path, RTLD_NOW | RTLD_LOCAL);
+      driver = dlopen(NULL, RTLD_NOW | RTLD_LOCAL);
+
       if (driver == NULL) {
          dl_error = dlerror();
          log_(_LOADER_DEBUG, "MESA-LOADER: failed to open %s: %s\n",
diff --git a/src/mesa/glapi/es2api/meson.build b/src/mesa/glapi/es2api/meson.build
index be999c1ab3b..a2e5775d226 100644
--- a/src/mesa/glapi/es2api/meson.build
+++ b/src/mesa/glapi/es2api/meson.build
@@ -22,10 +22,10 @@ gles2_def = custom_target(
   command : gen_vs_module_defs_normal_command,
 )
 
-libgles2 = shared_library(
+libgles2 = static_library(
   'GLESv2' + get_option('gles-lib-suffix'),
   ['libgles2_public.c', es2_glapi_mapi_tmp_h],
-  vs_module_defs : gles2_def,
+#  vs_module_defs : gles2_def,
   c_args : [
     c_msvc_compat_args,
     _es2_c_args,
@@ -36,9 +36,9 @@ libgles2 = shared_library(
   include_directories : [inc_mesa, inc_src, inc_include],
   link_with : shared_glapi_lib,
   dependencies : [dep_thread, dep_libdrm, dep_m, dep_dl, idep_mesautilc11],
-  soversion : host_machine.system() == 'windows' ? '' : '2',
-  version : '2.0.0',
-  darwin_versions : '3.0.0',
+#  soversion : host_machine.system() == 'windows' ? '' : '2',
+#  version : '2.0.0',
+#  darwin_versions : '3.0.0',
   name_prefix : host_machine.system() == 'windows' ? 'lib' : [],  # always use lib, but avoid warnings on !windows
   install : true,
 )
diff --git a/src/mesa/meson.build b/src/mesa/meson.build
index bfce1af9c1c..6ab4db39a83 100644
--- a/src/mesa/meson.build
+++ b/src/mesa/meson.build
@@ -465,5 +465,5 @@ libmesa = static_library(
   ],
   link_with : [libmesa_sse41],
   dependencies : [idep_libglsl, idep_nir, idep_vtn, idep_mesautil],
-  build_by_default : false,
+  build_by_default : true,
 )
diff --git a/src/util/cnd_monotonic.c b/src/util/cnd_monotonic.c
index 71d9ac0d247..4fbab736602 100644
--- a/src/util/cnd_monotonic.c
+++ b/src/util/cnd_monotonic.c
@@ -132,7 +132,7 @@ u_cnd_monotonic_timedwait(struct u_cnd_monotonic *cond, mtx_t *mtx,
    timespec_sub_saturate(&rel_time, abs_time, &now_time);
    int rt = pthread_cond_timedwait_relative_np(&cond->cond, mtx, &rel_time);
 #else
-   int rt = pthread_cond_timedwait(&cond->cond, mtx, abs_time);
+   int rt = pthread_cond_timedwait(&cond->cond, (pthread_mutex_t*)mtx, abs_time);
 #endif
    if (rt == ETIMEDOUT)
       return thrd_timedout;
@@ -151,6 +151,6 @@ u_cnd_monotonic_wait(struct u_cnd_monotonic *cond, mtx_t *mtx)
                             (PCRITICAL_SECTION)mtx, INFINITE);
    return thrd_success;
 #else
-   return (pthread_cond_wait(&cond->cond, mtx) == 0) ? thrd_success : thrd_error;
+   return (pthread_cond_wait(&cond->cond, (pthread_mutex_t*)mtx) == 0) ? thrd_success : thrd_error;
 #endif
 }
